src/plugins/table-of-contents/table-of-contents-info.js

```diff
@@ -51,14 +54,19 @@ class TableOfContentsInfo {
     return cloneDeep(content);
   }
 
-  // eslint-disable-next-line no-unused-vars
-  redactContent(content, _targetRoomId) {
-    return cloneDeep(content);
+  redactContent(content, targetRoomId) {
+    const redactedContent = cloneDeep(content);
+
+    redactedContent.text = this.gfm.redactCdnResources(
+      redactedContent.text,
+      url => couldAccessUrlFromRoom(url, targetRoomId) ? url : ''
+    );
+
+    return redactedContent;
```

Comment:
Woop woop, now we need tests here 😋

Reply:
Added (and also some for the markdown plugin which works exactly the same).

Reply:
👍